### ISSUE TITLE
[Tips] If NaN happen 

### ISSUE THREAD

**Issue by zsdonghao** [tensorlayer/hyperpose#50]
- 1) Use `tf.contrib.layers.xavier_initializer()`

- 2) Use gradient clipping

Change the following
```python
        opt = tf.train.MomentumOptimizer(lr_v, 0.9)
        train_op = opt.minimize(total_loss, global_step=global_step)
```
To:
```python
        tvars = tl.layers.get_variables_with_name('model', True, True)
        grads, _ = tf.clip_by_global_norm(tf.gradients(total_loss, tvars), 100)
        optimizer = tf.train.MomentumOptimizer(lr_v, 0.9)
        train_op = optimizer.apply_gradients(zip(grads, tvars), global_step=global_step)
```
